envpool/sokoban/sokoban_node.cc

```diff
@@ -184,4 +188,27 @@ bool SokobanNode::GetSuccessors(std::AStarSearch<SokobanNode>* astarsearch,
   return true;
 }
 
+bool SokobanNode::CornerWalls(const std::pair<int, int>& box) const {
+  bool found_wall = false;
```

Comment:
I find this function pretty hard to read, I'd have done something like:

```cc
for (delta) {
  new_found_wall = CheckWall(...)
  found_contiguous_wall = found_contiguous_wall || (found_wall && new_found_wall);
  found_wall = new_found_wall;
}
found_contiguous_wall = found_contiguous_wall || (found_wall && CheckWall(...));
```

but this is highly subjective so I'm happy to keep it as is

Reply:
thanks, this is much simpler and cleaner to look at!